File: week1/python/data_structures.py
# the primary data structures in Python are lists, tuples, sets, and dictionaries
# if youre familiar w/ the idea of arrays or sequences, lists are the similar here
# sets are just a list of unique values
# dictionaries would be like maps, hash maps, or key-value objects
# and tuples are a group of 'unchanging' values

listOfAnimalTypes = ['cats', 'dogs', 'lizards', 'dinos', 'hippos']

singleDino = {'name': 'T-Rex', 'aggression': 7, 'coolness': 10}
singleDino['name'] = 'Triceratops';

# unlike JS, Python dictionaries don't have the ability to create methods with them

# like PHP, Python makes heavy use of lists (like PHP's array)
# where a large part of your data manipulation, and built-in functions are for lists

# Python also have powerful list comprehension
# gives you the ability to generate lists from lists
# this is one of the bigger strengths in Python for data manipulation/generation

squares = [x**2 for x in range(10)]
# for x in range(10), gen a list of its squares

combination = [(x,y) for x in ['xs','s', 'm','l'] for y in ['blue shirt','red shirt', 'green shirt']]

# sets are handy for finding unique vals
# tuples are handy for related data
# dictionaries are handy for key-val pairs

File: week1/js/data_structures.js
// the primary data structures in JS are arrays and objects
// if youre familiar w/ the idea of lists or sequences, arrays are the similar here
// objects would be like maps, or hash maps

// there are diff ways to create an array or obj
// using 'literal' syntax is the most common
// [val, ...valN] for arrays
// { 'prop':val, ...propN } for objects

const listOfAnimalTypes = ['cats', 'dogs', 'lizards', 'dinos', 'hippos']

// JS is really flexible when it comes to working w/ objects
const singleDino = {
    'name': 'T-Rex',
    'aggression': 7,
    'coolness': 10
}
singleDino.name = 'Raptor';
singleDino['name'] = 'Triceratops';

const otherDino = {
    name: 'T-Rex',
    aggression: 7,
    coolness: 10,
    avgAggroCool: function() {
        return ((this.aggression + this.coolness) / 2);
    }
}

const aggroCoolCalc = otherDino.avgAggroCool();
console.log(aggroCoolCalc);

// in JS, everything is an obj
// obj are used for building things youll manipulate
// the bedrock for JS
